lang/java/avro/src/main/java/org/apache/avro/specific/SpecificData.java

```diff
@@ -122,6 +122,10 @@ public DatumWriter createDatumWriter(Schema schema) {
   /** Return the singleton instance. */
   public static SpecificData get() { return INSTANCE; }
 
+  private static final boolean USE_CUSTOM_CODERS
+    = Boolean.parseBoolean(System.getProperty("org.apache.avro.specific.use_custom_coders","false"));
+  public boolean useCustomCoders() { return USE_CUSTOM_CODERS; }
```

Comment:
useCustomCoders should probably be a settable member variable on each SpecificData instance, so folks can more easily switch it on and off without restarting the JVM.

Reply:
In https://github.com/rstata-projects/avro/commit/e81a5880be69aef5bffdcf38f7ae3491957a6a68, I changed this flag to be dynamically changable.

For the record, here's the roll-out plan I imagined for this feature (See [AVRO-2091](https://issues.apache.org/jira/browse/AVRO-2091)).  The initial release of this feature sets the `use_custom_coders` flag to `false` by default, to allow for some road testing.  The  release after that would set this flag to `true` by default, to allow for some more serious road testing.  The next release of Avro after that would then eliminate this flag altogether, under the assumption that the feature has been sufficiently tested and there's no longer a use for the old way of doing things.